src/main/java/org/generationcp/middleware/operation/builder/DataSetBuilder.java

```diff
@@ -209,7 +206,8 @@ public VariableTypeList filterVariables(final VariableTypeList variables, final
 		final VariableTypeList newList = new VariableTypeList();
 		if (variables != null && !variables.getVariableTypes().isEmpty()) {
 			for (final DMSVariableType variable : variables.getVariableTypes()) {
-				if (!filters.contains(variable.getId()) || this.DEFAULT_DATASET_COLUMNS.contains(variable.getId())) {
+				if (!filters.contains(variable.getId()) || variable.getId() == TermId.TRIAL_INSTANCE_FACTOR.getId()
+				|| (!variable.getVariableType().equals(VariableType.ENVIRONMENT_DETAIL) && !variable.getVariableType().equals(VariableType.STUDY_DETAIL))) {
```

Comment:
Hi @gelli-leafnode. Since we're already filtering out ENVIRONMENT and STUDY DETAILS, do we still need the filters list?
I think the variables shown in the table are VariableType.GERMPLASM_DESCRIPTOR, VariableType.TRAIT and TRIAL_INSTANCE_FACTOR. Can you double check and just filter those? Also check the scenario with Treatment Factors.

Reply:
Remove sibling variables from filters.